ocs_ci/ocs/resources/mcg.py

```diff
@@ -379,6 +381,23 @@ def _retrieve_reduction_data():
                 f"{total_size - total_reduced} bytes reduced out of {expected_reduction_in_bytes}."
             )
 
+    def determine_s3_endpoint(self):
+        """
+        Get internal mcg S3 endpoint if the cluster is in disconnected or proxy environment.
+        Get external endpoint otherwise.
+
+        Returns:
+            string: S3 endpoint URI
+        """
+        return (
+            self.s3_external_endpoint
+            if (
+                config.ENV_DATA.get("platform", "").lower()
+                in constants.HCI_PROVIDER_CLIENT_PLATFORMS
+            )
+            else self.s3_internal_endpoint
+        )
+
```

Comment:
1. Please update the Docstring
2. https://github.com/red-hat-storage/ocs-ci/pull/9733/commits/0faef55565b27346762f932b4c690a3169647ab9 changes `the self.s3_endpoint` class member to be the internal endpoint on most cases, which is a big change that we need to verify. 

While this endpoint should work on S3 operations that are made via the awscli pod which is within the cluster, some test cases are making s3 operations from the Jenkins agent via boto3 from outside of it. I fear this change will fail those kind of test cases.